sql/2022/media/image_decode_usage.sql

```diff
@@ -0,0 +1,47 @@
+#standardSQL
+# What percent of <img>s have decode=async?
+
+CREATE TEMPORARY FUNCTION get_decode_info(images_string STRING)
+RETURNS STRUCT<
+  total INT64,
+  decode_async INT64
+> LANGUAGE js AS '''
+let result = {};
+try {
+  let images = JSON.parse(images_string);
+  if (!Array.isArray(images)) {
+      return {};
+  }
+
+  result.total = images.length;
+  result.decode_async = 0;
+
+  for(let img of images) {
+      if(img.decoding === 'async'){
+          result.decode_async += 1
+      }
+  }
+} catch (e) {}
+return result;
+''';
+
+SELECT
+  client,
+  COUNT(0) AS pages_total,
+  SAFE_DIVIDE(COUNTIF(images_info.total > 0), COUNT(0)) AS pages_with_img_pct,
+  COUNTIF(images_info.decode_async > 0) AS pages_with_decode_async,
+  SAFE_DIVIDE(COUNTIF(images_info.decode_async > 0), COUNT(0)) AS pages_with_decode_async_pct,
+  SUM(images_info.total) AS img_total,
+  SUM(images_info.decode_async) AS imgs_with_decode_async,
+  SAFE_DIVIDE(SUM(images_info.decode_async), SUM(images_info.total)) AS imgs_with_decode_async_pct
+FROM (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    url,
+    get_decode_info(JSON_EXTRACT_SCALAR(payload, '$._Images')) AS images_info
+  FROM
+    `httparchive.pages.2022_07_01_*`)
```

Comment:
```suggestion
    `httparchive.pages.2022_06_01_*`)
```

Reply:
Leaving this one unresolved as a reminder for you to rerun the query with June data